Move tests_common from "dev" to top-level.

potiuk opened the pull request:
Follow-up after #42505 fixing teething problem with tests_common.

Originally in #42505 common test code was moved to "dev" folder, but the "dev" folder is really dedicated to "build" scripts and the problem with moving "tests_common" to the folder was that the whole "dev" folder is replaced (for non-committer PRs) with the content from the target branch.

This is done for security reasons, because we can accidentally use any of the scripts from dev in the CI build scripts and we might not notice, which will open us to a security issue where a file in "dev" coming from PR could be accidentally executed during the "pull_request_target" workflow - which would expose our secrets and GitHub Package write permissions to a contributor coming from a fork.

This change moves the files, fixes pre-commit specification and docs, also fixes a number of "doc" issues detected by "ruff" in the tests_common folder as they were detected after the move. The tests_common folder is added to folders mounted when breeze is executed with local folders mounted (in order to avoid accidental mounting of randomly generated files to inside the breeze container).

All imports for the common tests were updated to reflect this move.

<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->



<!-- Please keep an empty line above the dashes. -->
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

ashb commented on providers/tests/alibaba/cloud/log/test_oss_task_handler.py:
```diff
@@ -22,14 +22,13 @@
 
 import pytest
+from tests_common.test_utils.config import conf_vars
+from tests_common.test_utils.db import clear_db_dags, clear_db_runs
```
Please add this to the ruff config section in pyproject.toml so that it knows this is a "testing" group.

This section on ~L350
```toml
# Make sure we put the "dev" imports at the end, not as a third-party module
[tool.ruff.lint.isort.sections]
testing = ["dev", "providers.tests", "task_sdk.tests"]
```

potiuk replied:
Very good point. That will limit a number of changes! Thanks!

potiuk replied:
Fixed. Will push it shortly.

jscheffl commented on providers/tests/system/airbyte/example_airbyte_trigger_job.py:
```diff
@@ -61,7 +61,13 @@
+test_run = get_test_run(dag)
+
+
+from tests_common.test_utils.system_tests import get_test_run  # noqa: E402
```
Why do you import two times here? Can the E402 be removed or is this a specific to system test?

jscheffl replied:
Same below repeating... so probably it is a feature, not a bug? :-D

potiuk replied:
VERY good find.

 I think it's result of pre-commit run before/after the change. Pre-commit added the lines thinking it's not there before import has been updated in code - but then search-replace replaced the old import with the new one - making it into a duplicate.

I will fix it

potiuk replied:
Yep. That was it. Fixed it - will push it shortly.

potiuk replied:
BTW. That was also a reason for the doc-build errors - get_test_run was seen as duplicate identifer by sphinx :D. Killing two birds with the same stone :)

jscheffl reviewed (approved):
Some sanity question - a LOT of copy & paste... smells some decorators or generaotrs could remove a lot of duplication?

Crossing fingers you get it green.

For easier review... might be better to make some smaller incremental moves w/o any content changes, then git diff is faster to scroll through :-D

potiuk committed:
Move tests common without changes

potiuk committed:
Fix docstrings in tests_common

potiuk committed:
Move tests_common from "dev" to top-level.

Follow-up after #42505 fixing teething problem with tests_common.

Originally in #42505 common test code was moved to "dev" folder, but
the "dev" folder is really dedicated to "build" scripts and the
problem with moving "tests_common" to the folder was that the
whole "dev" folder is replaced (for non-committer PRs) with
the content from the target branch.

This is done for security reasons, because we can accidentally
use any of the scripts from dev in the CI build scripts and we
might not notice, which will open us to a security issue where
a file in "dev" coming from PR could be accidentally executed
during the "pull_request_target" workflow - which would expose
our secrets and GitHub Package write permissions to a
contributor coming from a fork.

This change moves the files, fixes pre-commit specification and
docs, also fixes a number of "doc" issues detected by "ruff" in
the tests_common folder as they were detected after the move.
The tests_common folder is added to folders mounted when
breeze is executed with local folders mounted (in order to
avoid accidental mounting of randomly generated files to
inside the breeze container).

All imports for the common tests were updated to reflect this
move.

jscheffl commented on airflow/models/dag.py:
```diff
@@ -1386,6 +1386,7 @@ def get_template_env(self, *, force_sandboxed: bool = False) -> jinja2.Environme
             "extensions": ["jinja2.ext.do"],
             "cache_size": 0,
         }
+        self.log.warning("SEARCHPATH: %s", searchpath)
```
Is this a testing leftover or what is this change for?

potiuk replied:
aaaaaaaaaa.... Stupid me. 

Glad that the PR is now "somewhat" reviewable". And that you did it ! 

Let me re-review it myself before next push.
